doc/src/set.rst

```diff
@@ -237,24 +245,32 @@ from a file.
    such as the molecule ID, then the floating point value is truncated to
    its integer portion, e.g. a value of 2.6 would become 2.
 
-Keyword *type* sets the atom type for all selected atoms.  The
-specified value must be from 1 to ntypes, where ntypes was set by the
-:doc:`create_box <create_box>` command or the *atom types* field in the
+.. versionchanged:: 28Mar2023
+
+   Support for type labels was added for setting atom, bond, angle,
+   dihedral, and improper types
+
+Keyword *type* sets the atom type or atom type label for all selected atoms.
+For atom type, specified value must be from 1 to ntypes, where ntypes was set
+by the :doc:`create_box <create_box>` command or the *atom types* field in the
 header of the data file read by the :doc:`read_data <read_data>`
-command.
+command. In the case of the specified type label, it must have been defined
+before use via an alphanumeric type label. See the
+:doc:`Howto type labels <Howto_type_labels>` doc page for the allowed syntax
+of type labels and a general discussion of how type labels can be used.
 
-Keyword *type/fraction* sets the atom type for a fraction of the
-selected atoms.  The actual number of atoms changed is not guaranteed
+Keyword *type/fraction* sets the atom type or atom type label for a fraction
```

Comment:
You don't need to add "or atom type label" here. The command does not set the atom type label, it still just sets the atom type. This PR simply allows type labels to set the atom type.